### PR TITLE
ACS-934: Add helm option OOI service component & ADW env vars (including example doc)

### PR DESCRIPTION
**Pull request by KikoUK**
- Add both OOI Service and ADW env variables to the ACS Deployment helm charts (as an option)
- Document in Markdown as an Example extension and add link to new page within Deploy section

### CODE REVIEW

**Comment by montgolfiere** [comment]
@KikoUK I see description above mentions new md file. Can you please add the draft to this PR ?

**Comment by KikoUK** [comment]
> @KikoUK I see description above mentions new md file. Can you please add the draft to this PR ?

The md is still WIP, I was going to add it after testing the changes in the PR

File: helm/alfresco-content-services/values.yaml
Comment (montgolfiere) [inline helm/alfresco-content-services/values.yaml:473]: Remove hard-coded values ...

**Review by montgolfiere — CHANGES_REQUESTED** [review]
Early comments added to Draft PR.

Will review again once we have the markdown and hard-coded values removed.

File: helm/alfresco-content-services/values.yaml
Comment (montgolfiere) [inline helm/alfresco-content-services/values.yaml:469]: This will need to be 'false' by default ?

File: helm/alfresco-content-services/values.yaml
Comment (montgolfiere) [inline helm/alfresco-content-services/values.yaml:474]: This doesn't look right ? Should the curly braces be here ?

Comment (montgolfiere) [reply helm/alfresco-content-services/values.yaml]: I think I've fixed this now.

**Comment by montgolfiere** [comment]
I've made a few more commits to fix (helm) lint errors, use tcp probes, ensure OOI Service starts & env vars get passed to ADW pod and also added docs. Hence, marking as "Ready for review".

Note: we still need to sanity check in Docker Desktop &/or remote EKS deployment with configured integration.

File: helm/alfresco-content-services/templates/config-ooi-service.yaml
Comment (gavincornwell) [inline helm/alfresco-content-services/templates/config-ooi-service.yaml:1]: Is OOI available for Community too?

If not, it might also be worth adding a check for Enterprise edition:

`{{- if eq .Values.repository.edition "Enterprise" }}`

as used in helm/alfresco-content-services/templates/deployment-transform-router.yaml

**Review by gavincornwell — CHANGES_REQUESTED** [review]
Left a couple of inline comments/questions

File: helm/alfresco-content-services/requirements.yaml
Comment (montgolfiere) [reply helm/alfresco-content-services/requirements.yaml:12]: Thanks.

A few reasons. Mainly to make it shorter for --set (see docs above) and also if ever used via config map, I think it fails the helm lint due to "-" characters. I'm ok to rename back for now but may need to revisit in the future.

File: helm/alfresco-content-services/templates/config-ooi-service.yaml
Comment (montgolfiere) [reply helm/alfresco-content-services/templates/config-ooi-service.yaml:1]: No, not Community, sorry. Note: ADW and OOI are currently Enterprise-only.

File: helm/alfresco-content-services/requirements.yaml
Comment (montgolfiere) [reply helm/alfresco-content-services/requirements.yaml]: I've renamed back from "adw" to "alfresco-digital-workspace" (and updated example doc accordingly).

**Review by montgolfiere — APPROVED** [review]

**Comment by montgolfiere** [comment]
LGTM - subject to testing

File: helm/alfresco-content-services/requirements.yaml
Comment (gavincornwell) [reply helm/alfresco-content-services/requirements.yaml]: Thanks @montgolfiere. There are other items with dashes in the name so it will be a widespread problem if "-" is not allowed :-(

**Comment by gavincornwell** [comment]
LGTM as well once the extra checks for Enterprise have been added.

**Comment by montgolfiere** [comment]
@gavincornwell Thanks. I've made the extra change (to also check for Enterprise when enabling OOI Service).

The merge is pending testing of the config options. At the moment, not yet sure if the ADW env vars are being picked-up by the ADW container as expected (although I can see them in the "env" section of the running pod).

File: helm/alfresco-content-services/templates/config-ooi-service.yaml
Comment (montgolfiere) [reply helm/alfresco-content-services/templates/config-ooi-service.yaml:1]: Added the extra check.

File: docs/helm/examples/with-ooi.md
Comment (mauriziovitale) [inline docs/helm/examples/with-ooi.md:44]: the value should be a boolean not a string

File: docs/helm/examples/with-ooi.md
Comment (mauriziovitale) [inline docs/helm/examples/with-ooi.md:54]: same here true not 'true'

File: docs/helm/examples/with-ooi.md
Comment (montgolfiere) [reply docs/helm/examples/with-ooi.md]: Unfortunately fails with "helm install" with error documented below. I've tried all combinations of:

--set ...=true
--set...='true'
--set...="true"

However after some searching, I may have found a helm workaround, ie. use:

--set-string alfresco-digital-workspace.env.APP_CONFIG_PLUGIN_MICROSOFT_ONLINE="true" \

File: docs/helm/examples/with-ooi.md
Comment (montgolfiere) [reply docs/helm/examples/with-ooi.md]: See above fix (to use --set-string). I've also changed this note.

**Comment by montgolfiere** [comment]
I've now fixed the OOI Service including the new ingress. I've done a quick sanity check and could deploy and run Office Online Integration ([Microsoft 365 Connector](https://docs.alfresco.com/officeonline/concepts/office-online-intro.html)) in Kubernetes.

I believe this PR is now ready for any final review comments & merge (cc @PunchedChimera @KikoUK @eknizat).

**Comment by montgolfiere** [comment]
It would be good to get at least one more review & approve. I/we can then merge, thanks. 

cc @gavincornwell @sergiuv2020 @nilsSure

**Review by PunchedChimera — APPROVED** [review]

**Comment by montgolfiere** [comment]
Master is green again (as per OPSEXP-61). Hence, will merge.

**Pull request merged**
